spyder/plugins/completion/plugin.py

```diff
@@ -170,19 +193,43 @@ def gather_and_send(self, principal_source, response_instance, req_type,
             # removed before the response can be processed.
             pass
 
-    def send_request(self, language, req_type, req, req_id=None):
+    def _is_client_running(self, name):
+        status = self.clients.get(name, {}).get('status', self.STOPPED)
+        return status == self.RUNNING
+
+    def send_request(self, language, req_type, req):
         req_id = self.req_id
-        for client_name in self.clients:
+
+        kite = KiteCompletionPlugin.COMPLETION_CLIENT_NAME
+        lsp = LanguageServerPlugin.COMPLETION_CLIENT_NAME
+        fallback = FallbackPlugin.COMPLETION_CLIENT_NAME
+
+        client_names = []
+        if req_type not in self.MUTEX_REQUEST_TYPES:
+            client_names.append(kite)
+            client_names.append(lsp)
+        elif (self.get_option('enable_kite', True) and
+              self._is_client_running(kite)):
+            client_names.append(kite)
+        else:
+            client_names.append(lsp)
+
+        if (self.get_option('enable_fallback', True) and
+                self._is_client_running(fallback)):
+            client_names.append(fallback)
+
+        for client_name in client_names:
             client_info = self.clients[client_name]
```

Comment:
`test_lsp_splitter_close` is failing at this point because it seems the Kite client is not added to the request that test is checking.